CMakeLists.txt

```diff
@@ -20,6 +20,17 @@ set(CMAKE_CXX_EXTENSIONS ON)
 # Must be at global scope, otherwise breaks -DPLAYER_BUILD_LIBLCF (see CMP0077)
 option(BUILD_SHARED_LIBS "Build shared easyrpg_libretro core" ON)
 
+# ninja/parallel builds disable output colors, forcing them here
+option(CMAKE_FORCE_COLORS "Always produce ANSI-colored output" OFF)
+if(${CMAKE_FORCE_COLORS})
+	if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+		add_compile_options("-fdiagnostics-color=always")
+	elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+		add_compile_options("-fcolor-diagnostics")
+	endif()
+	set(DOCTEST_COLOROPT "--force-colors")
+endif()
```

Comment:
FYI: CMake just added https://github.com/Kitware/CMake/blob/cbd36eac23628f74ae3b50f9d1b8a0478c3e317a/Help/release/dev/color-diagnostics.rst#id1  which introduces ``CMAKE_COLOR_DIAGNOSTICS``.

Not usable yet as this is a dev feature.

Reply:
I guess I can wait until 3.24, other idea is to wrap this into a CMAKE_VERSION conditional and use the official name, so it will work for older versions too. At least for doctest there needs to be a way to set it.

Reply:
Maybe keep your code and call it ``CMAKE_COLOR_DIAGNOSTICS`` and wrap it already in a version check so the var name stays the same across versions.

Reply:
This is exactly what I meant :D

Reply:
applied.